metrics/opencensus_exporter.go

```diff
@@ -35,6 +35,9 @@ func newOpenCensusExporter(config *metricsConfig, logger *zap.SugaredLogger) (vi
 	if config.collectorAddress != "" {
 		opts = append(opts, ocagent.WithAddress(config.collectorAddress))
 	}
+	if config.component != "" {
```

Comment:
Do we need `config.domain` as well?

How does this match with `WithServiceName` on line 34?

Stackdriver sets the metric prefix to `path.Join(mc.domain, mc.component) on line 256 of `config.go`.

Reply:
It looks like `WithServiceName` is used to fill in [`ServiceInfo` on the Node information representing the system that's reporting the metrics](https://github.com/census-instrumentation/opencensus-proto/blob/master/src/opencensus/proto/agent/common/v1/common.proto#L44).

Reply:
We don't want to add domain here, which is a stackdriver specific here.  If we add it here, all other metric backends in the opentelemetry metrics collector service will get the domain prefix, which is undesired.

Instead the domain prefix will be added by the stackdriver exporter  in opentelemetry metrics collector service.

Reply:
We could use the `ServiceInfo`, but then we need to modify the exporters in opentelemetry metrics collector service to be aware of it and add it as the prefix.

The prefix here is more straightforward.

Reply:
I agree that adding the prefix here:

1. Makes sense
2. Is more straightforward than extracting from `ServiceInfo`

My question was around avoiding collisions in the resource names; right now Stackdriver (which has a resource-path name) names the exports something like:

* `knative.dev/serving/autoscaler/actual_pods`
* `custom.googleapis.com/knative.dev/autoscaler/not_ready_pods`

I'm wondering if the `serving` (and `knative.dev`) part of the first name should be passed along via OpenCensus as well as the `autoscaler` part. Passing along those first parts of the domain makes it easier to avoid namespace collisions (for example, if there were "controller" components in both serving and eventing).

Reply:
In particular, I've found a metrics named collision between serving and eventing which is resolved via the `mc.domain` field, which is extracted from the `METRICS_DOMAIN` environment variable:

* Serving: [`config/core/deployments/controller.yaml`](https://github.com/knative/serving/blob/master/config/core/deployments/controller.yaml#L60) with name [`controller`](https://github.com/knative/serving/blob/master/cmd/controller/main.go#L45)
* Eventing: [`config/core/deployments/controller.yaml`](https://github.com/knative/eventing/blob/master/config/core/deployments/controller.yaml#L55) with name [`controller`](https://github.com/knative/eventing/blob/master/cmd/controller/main.go#L37)

I'd prefer to clean up the `METRICS_DOMAIN` in serving to all be under the same domain, but we'll need to see if that's breaking for anyone.